src/image_builder.py

```diff
@@ -154,109 +156,198 @@ def image_builder(buildspec):
             FORMATTER.title(image.name)
-            FORMATTER.table(image.summary.items())
-
-        FORMATTER.title("Errors")
-        is_any_build_failed = False
-        is_any_build_failed_size_limit = False
-        for image in IMAGES:
-            if image.build_status == constants.FAIL:
-                FORMATTER.title(image.name)
-                FORMATTER.print_lines(image.log[-10:])
-                is_any_build_failed = True
-            else:
-                if image.build_status == constants.FAIL_IMAGE_SIZE_LIMIT:
-                    is_any_build_failed_size_limit = True
-        if is_any_build_failed:
-            raise Exception("Build failed")
+            FORMATTER.print_lines(image.log[-10:])
+            is_any_build_failed = True
+        else:
+            if image.build_status == constants.FAIL_IMAGE_SIZE_LIMIT:
+                is_any_build_failed_size_limit = True
+    if is_any_build_failed:
+        raise Exception("Build failed")
+    else:
+        if is_any_build_failed_size_limit:
+            FORMATTER.print("Build failed. Image size limit breached.")
         else:
-            if is_any_build_failed_size_limit:
-                FORMATTER.print("Build failed. Image size limit breached.")
+            FORMATTER.print("No errors")
+    return is_any_build_failed, is_any_build_failed_size_limit
+
+def upload_metrics(images, BUILDSPEC, is_any_build_failed, is_any_build_failed_size_limit):
+
+    FORMATTER.title("Uploading Metrics")
+    is_any_build_failed = False
+    is_any_build_failed_size_limit = False
+    metrics = Metrics(
+        context=constants.BUILD_CONTEXT,
+        region=BUILDSPEC["region"],
+        namespace=constants.METRICS_NAMESPACE,
+    )
+    for image in images:
+        try:
+            metrics.push_image_metrics(image)
+        except Exception as e:
+            if is_any_build_failed or is_any_build_failed_size_limit:
+                raise Exception(f"Build failed.{e}")
             else:
-                FORMATTER.print("No errors")
+                raise Exception(f"Build passed. {e}")
 
-        FORMATTER.title("Uploading Metrics")
-        metrics = Metrics(
-            context=constants.BUILD_CONTEXT,
-            region=BUILDSPEC["region"],
-            namespace=constants.METRICS_NAMESPACE,
-        )
-        for image in IMAGES:
-            try:
-                metrics.push_image_metrics(image)
-            except Exception as e:
-                if is_any_build_failed or is_any_build_failed_size_limit:
-                    raise Exception(f"Build failed.{e}")
-                else:
-                    raise Exception(f"Build passed. {e}")
+    if is_any_build_failed_size_limit:
+        raise Exception("Build failed because of file limit")
 
-        if is_any_build_failed_size_limit:
-            raise Exception("Build failed because of file limit")
+    FORMATTER.separator()
 
-        FORMATTER.separator()
+def build_images(images, make_dummy_boto_client=False):
```

Comment:
Please add a docstring to this function.

Reply:
Done.